### PR TITLE
Update go modules including replacing istio/pkg

### PR DESCRIPTION
**Pull request by ericvn**
Noting we should update [google.golang.org/grpc](http://google.golang.org/grpc) for a vulnerability I decided to update all the modules as we haven't done that in this repo in a while as well as change from the deprecated istio/pkg.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,78 +1,113 @@
 module istio.io/bots
 
-go 1.18
+go 1.21
+
+toolchain go1.21.4
```
Comment (howardjohn) [inline go.mod:5]: Why do we need this? Does this enforce everyone is using 1.21.4?

Comment (ericvn) [reply go.mod:5]: Interestingly, that came in as part of the `go mod tidy`. Will remove that.

Still chasing why the license files generated from my `make gen` locally are different than what the pipeline generates. This is the second item I've found is different, the first being sort, when using the build-tools image on my local machine vs the pipeline.

**Comment by ericvn** [comment]
Still chasing the license file difference between the pipeline and on my Mac.

**Review by howardjohn — APPROVED** [review]

**Comment by ericvn** [comment]
I note that cpuid is not returned as a module when running locally:
```
> go list -mod=readonly -deps -test -json ./... | grep cpuid  
> 
```

**Pull request merged**